### PR TITLE
Add the SMBus Process Call and Block Process Call commands.

### PR DESCRIPTION
**Pull request by jmeurin**
Signed-off-by: <JeanMarc.Eurin@gmail.com>

### CODE REVIEW

File: smbus_rw.c [hunk 1]
```diff
@@ -245,124 +364,7 @@ smbus_read_op(struct smbus_op_params *params, const struct smbus_op *op)
...
-			fprintf(stderr, "%s: isn't 0 or 1\n", arg);
-			return -1;
-		}
-		params->data.fixed.u8 = ldata;
-		break;
-	case SMBUS_SIZE_BYTE:
-	case SMBUS_SIZE_8:
-		ldata = strtoul(arg, &end, 0);
-		if (ismaxortrailingjunk(arg, end, ldata))
-			return -1;
```
Comment (adurbin) [inline smbus_rw.c]: Why did you change the formatting of all this code?

File: smbus_rw.c [hunk 2]
```diff
@@ -137,6 +141,119 @@ parse_uint8_hex(const char *arg, uint8_t *ret)
...
+		if (istrailingjunk(arg, end)) return -1;
+		params->data.fixed.u32 = ldata;
+		break;
+	case SMBUS_SIZE_64:
+		ldata = strtoull(arg, &end, 0);
+		if (istrailingjunk(arg, end)) return -1;
+		params->data.fixed.u64 = ldata;
+		break;
+	case SMBUS_SIZE_BLOCK:
+	case SMBUS_BLOCK_PROC_CALL: {
```
Comment (adurbin) [inline smbus_rw.c]: The only thing that should have changed in this function is adding 2 new case statements. Nothing else.

File: smbus_rw.c [hunk 3]
```diff
@@ -460,6 +462,81 @@ smbus_write_op(struct smbus_op_params *params, const struct smbus_op *op)
...
+	int ret;
+	struct smbus_op_params params;
+	const struct smbus_op *op = (const struct smbus_op *)info->privdata;
+
+	if (smbus_prologue(argv, &params, op) < 0) {
+		return -1;
+	}
+
+	if (parse_io_width(argv[4], &params, op) < 0) {
+		fprintf(stderr, "%s: %s: invalid value to write\n", argv[0], argv[4]);
```
Comment (adurbin) [inline smbus_rw.c]: line is over 80 cols

File: smbus_rw.c [hunk 4]
```diff
@@ -137,6 +141,119 @@ parse_uint8_hex(const char *arg, uint8_t *ret)
...
+		break;
+	case SMBUS_SIZE_64:
+		printf("0x%016llX\n", (long long)data->fixed.u64);
+		break;
+	case SMBUS_SIZE_BLOCK: {
+		int i;
+
+		if (read_block_size > I2C_SMBUS_BLOCK_MAX) /* sanity */
+			read_block_size = I2C_SMBUS_BLOCK_MAX;
+		printf("0x");
```
Comment (adurbin) [inline smbus_rw.c]: So you've added a 0x now? Why are there all kinds of unrelated changes in this patch? And nothing is listed in the patch description.

File: smbus_rw.c [hunk 5]
```diff
@@ -137,6 +141,119 @@ parse_uint8_hex(const char *arg, uint8_t *ret)
...
+	case SMBUS_SIZE_16:
+		printf("0x%04X\n", data->fixed.u16);
+		break;
+	case SMBUS_SIZE_32:
+		printf("0x%08X\n", data->fixed.u32);
+		break;
+	case SMBUS_SIZE_64:
+		printf("0x%016llX\n", (long long)data->fixed.u64);
+		break;
+	case SMBUS_SIZE_BLOCK: {
```
Comment (adurbin) [inline smbus_rw.c]: formatting has changed.

**Comment by adurbin** [comment]
https://github.com/adurbin/iotools/commit/118237d2aa35fd321c7109076c4efc99e6145e8c

File: smbus_rw.c (hunk 2 above)
Comment (jmeurin) [inline smbus_rw.c]: I moved it up to have all the parse_ functions together.

File: smbus_rw.c (hunk 4 above)
Comment (jmeurin) [inline smbus_rw.c]: This made it consistent with the other read sizes.  Github is such a pain that splitting those in different commits is not worth it, IMHO.

File: smbus_rw.c (hunk 3 above)
Comment (jmeurin) [inline smbus_rw.c]: Sorry, missed that one.
